### ISSUE TITLE
LSTM - forecasting time series at t+1

### ISSUE THREAD

**Issue by philipperemy** [fchollet/keras#2463]
Hello Keras community!

It's been a few I'm dev on Keras but I'm encountering a small problem for which you might know.
I have 1000 1-D time series, each one of length 101.
I'm interesting in understanding the mechanism t -> t+1 based on the sequence up to time t. I, thus, have 100 corresponding mapping t->t+1.

I'm using a simple LSTM with:
Input `(nb_samples, timesteps, input_dim) = (1000, 100, 1)`

What should be my output? Is this correct:
Output `(nb_samples, timesteps, input_dim) = (1000, 100, 1)`

Also can you give me a simple LSTM that could do this one-to-one mapping.

Thanks!

**Comment by jllombart** [comment]
I'm not sure but you may expect as output (nb_samples, timesteps, number_of_hidden_units_in_LSTM)
to get this you have to add a dense softmax in which you perform the output as your output labels are.
You can see the lstm_text_generation.py in which is used something like you want.

**Comment by fluency03** [comment]
I think this [lstm_text_generation.py](https://github.com/fchollet/keras/blob/master/examples/lstm_text_generation.py) example is a proper one you can look into.

**Comment by ashwin123** [comment]
You can probably try the following architecture : 
```
model = Sequential()
model.add(LSTM(1,batch_input_shape=(nb_samples,timesteps, input_dim),return_sequences=True))
```
